From 8b95cda16dfec4b8a26385b745e5929c322e744c Mon Sep 17 00:00:00 2001
From: Zidan Wang <zidan.wang@freescale.com>
Date: Fri, 25 Sep 2015 14:27:00 +0800
Subject: [PATCH 0516/5242] MLK-11628 ASoC: fsl_sai: add initial value for
 is_slave_mode

commit  af6556468852ed8ef161e041ba2b3fadd415c2fc from
https://source.codeaurora.org/external/imx/linux-imx.git

After playback audio with sai<->wm8960 sound card, is_slave_mode
will be set, but it will not be cleared. So playback audio with
sai<->sii902x sound card will have no voice.

Signed-off-by: Zidan Wang <zidan.wang@freescale.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 sound/soc/fsl/fsl_sai.c |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/sound/soc/fsl/fsl_sai.c b/sound/soc/fsl/fsl_sai.c
index d5e9435..fde6af6 100644
--- a/sound/soc/fsl/fsl_sai.c
+++ b/sound/soc/fsl/fsl_sai.c
@@ -265,6 +265,8 @@ static int fsl_sai_set_dai_fmt_tr(struct snd_soc_dai *cpu_dai,
 		return -EINVAL;
 	}
 
+	sai->is_slave_mode = false;
+
 	/* DAI clock master masks */
 	switch (fmt & SND_SOC_DAIFMT_MASTER_MASK) {
 	case SND_SOC_DAIFMT_CBS_CFS:
-- 
1.7.9.5

